tests/test_utils.sh

```diff
@@ -710,6 +710,51 @@ function test_is_function {
   unalias not_my_function
 }
 
+function test_hash_color {
+    PS1="$ "
+    lp_activate --no-config # For having _lp_foreground
+    _lp_hash_color "Debug"
+    assertContains "$lp_hash_color" "Debug"
+    # FIXME How to test color?
+}
+
+function test_join {
+    _lp_join "_" "A" "B " " " " C" " D " "EE"
+    assertEquals "A_B _ _ C_ D _EE" "$lp_join"
+
+    _lp_join "-" "A"
+    assertEquals "A" "$lp_join"
+
+    _lp_join "+" ""
+    assertEquals "" "$lp_join"
+
+    typeset -a arr
+    arr=(1 2 3)
+    _lp_join "/" "${arr[@]}"
+    assertEquals "1/2/3" "$lp_join"
+}
+
+function test_grep_fields {
+    filename=$(mktemp)
+    echo -e "key1:value1\nkey2:value2" > "$filename"
```

Comment:
`echo -e` is brittle, prefer `printf` everywhere.
```suggestion
    printf 'key1:value1\nkey2:value2' > "$filename"
```